api/src/test/java/com/epam/pipeline/controller/cluster/ClusterControllerTest.java

```diff
@@ -0,0 +1,334 @@
+        params.add("regionId", "1");
+        params.add("spot", "false");
+
+        final AllowedInstanceAndPriceTypes allowedInstanceAndPriceTypes =
+                NodeCreatorUtils.getDefaultAllowedInstanceAndPriceTypes();
+
+        Mockito.doReturn(allowedInstanceAndPriceTypes).when(mockClusterApiService)
+                .getAllowedInstanceAndPriceTypes(ID, ID, false);
+
+        final MvcResult mvcResult = performRequest(get(LOAD_ALLOWED_INSTANCE_TYPES_URL).params(params));
+
+        Mockito.verify(mockClusterApiService).getAllowedInstanceAndPriceTypes(ID, ID, false);
+
+        assertResponse(mvcResult, allowedInstanceAndPriceTypes, NodeCreatorUtils.ALLOWED_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailGetNodeUsageStatisticsForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(NODE_USAGE_URL, NAME)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetNodeUsageStatistics() throws Exception {
+        final List<MonitoringStats> monitoringStats = Collections.singletonList(new MonitoringStats());
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("from", FROM_STRING.replace('T', ' '));
+        params.add("to", TO_STRING.replace('T', ' '));
+
+        Mockito.doReturn(monitoringStats).when(mockClusterApiService)
+                .getStatsForNode(NAME, from, to);
+
+        final MvcResult mvcResult = performRequest(get(String.format(NODE_USAGE_URL, NAME)).params(params));
+
+        Mockito.verify(mockClusterApiService).getStatsForNode(NAME, from, to);
+
+        assertResponse(mvcResult, monitoringStats, NodeCreatorUtils.MONITORING_STATS_TYPE);
+    }
+
+    @Test
+    public void shouldFailDownloadNodeUsageStatisticsReportForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(NODE_STATISTICS_URL, NAME)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDownloadNodeUsageStatisticsReport() throws Exception {
+        final InputStream inputStream = new ByteArrayInputStream(TEST_DATA.getBytes());
+
+        Mockito.doReturn(inputStream).when(mockClusterApiService)
+                .getUsageStatisticsFile(NAME, from, to, Duration.ofHours(1));
+
+        final MvcResult mvcResult = mvc().perform(get(String.format(NODE_STATISTICS_URL, NAME))
+                .servletPath(SERVLET_PATH)
+                .contentType("application/octet-stream")
+                .param("from", FROM_STRING.replace('T', ' '))
+                .param("to", TO_STRING.replace('T', ' '))
+                .param("interval", "PT1H"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType("application/octet-stream"))
+                .andReturn();
+
+        Mockito.verify(mockClusterApiService).getUsageStatisticsFile(NAME, from, to, Duration.ofHours(1));
+
+        String actualResponseData = mvcResult.getResponse().getContentAsString();
```

Comment:
Could you please make this field final or call it directly in `Assert.assertEquals`?